### PR TITLE
Update Readme.md and Makefile to new syntax. Upgrade pre-commit hooks to v0.1.4

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,21 +42,14 @@ In contrast to the plain `aws_iam_role` resource this module simplifies adding I
   Create an inline IAM policy, Attach custom or AWS managed policies. Create an IAM instance profile.
 
 ## Getting Started
-Basic usage for granting an AWS Account with Account ID `123456789012` access to assume a role that grants
-full access to AWS Simple Storage Service (S3)
+
+Basic usage:
```
Comment (mariux) [inline README.md:46]: please do not change already written documentation back to the template

File: README.md [hunk 2]
```diff
@@ -31,21 +42,14 @@ In contrast to the plain `aws_iam_role` resource this module simplifies adding I
...
-
-  assume_role_principals = [
-    {
-      type        = "AWS"
-      identifiers = ["arn:aws:iam::123456789012:root"]
-    }
-  ]
+  name = "s3-full-access"
 
   policy_statements = [
```
Comment (mariux) [inline README.md:54]: revert

File: README.md (hunk 1 above)
Comment (soerenmartius) [reply README.md]: resolved in d93f36f

File: README.md (hunk 2 above)
Comment (soerenmartius) [reply README.md]: resolved in d93f36f

**Comment by soerenmartius** [comment]
> Please revert redundant links to links with same http markdown handles pure links just fine by default..

I would rather stick to GH flavored markdown: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md034---bare-url-used
https://github.github.com/gfm/#links

**Comment by mariux** [comment]
> > Please revert redundant links to links with same http markdown handles pure links just fine by default..
> 
> I would rather stick to GH flavored markdown: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md034---bare-url-used
> https://github.github.com/gfm/#links

See "Automatic linking for URLs" in https://guides.github.com/features/mastering-markdown/ official github guide. you reference to the implementation details when writing github-md parsers... we are users..  but for your reference see also https://github.github.com/gfm/#example-619 ;)

**Comment by soerenmartius** [comment]
> > > Please revert redundant links to links with same http markdown handles pure links just fine by default..
> > 
> > 
> > I would rather stick to GH flavored markdown: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md034---bare-url-used
> > https://github.github.com/gfm/#links
> 
> See "Automatic linking for URLs" in https://guides.github.com/features/mastering-markdown/ official github guide. you reference to the implementation details when writing github-md parsers... we are users.. but for your reference see also https://github.github.com/gfm/#example-619 ;)

resolved in 08a5c95

File: README.md [hunk 3]
```diff
@@ -222,17 +245,20 @@ for details and use-cases.
...
 
 ## External Documentation
+
 - AWS Documentation IAM:
-  - Roles: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
-  - Policies: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html
-  - Instance Profile: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html
+  - Roles: [https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html)
+  - Policies: [https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html)
+  - Instance Profile: [https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html)
```
Comment (mariux) [inline README.md:261]: please remove redundant link naming

Comment (soerenmartius) [reply README.md]: resolved in 4ab8a34

**Comment by soerenmartius** [comment]
> lgtm..
> 
> please clean git history creating at least two different commits for build-system updates and documentation updates as both are not related...
> 
> and rename the PR accordingly as the title describes a completely different scope.
> 
> thx ;)

Cleaned up the history so we have two commits only. Changed the title of this PR also.

**Pull request merged**